ui/components/integrations/security-hub/security-hub-integration-form.tsx

```diff
@@ -0,0 +1,501 @@
+      if (!values.use_custom_credentials) {
+        // Use provider credentials - send empty object
+        formData.append("credentials", JSON.stringify({}));
+      } else {
+        // Use custom credentials
+        const credentials = buildCredentials(values);
+        formData.append("credentials", JSON.stringify(credentials));
+      }
+    } else {
+      const configuration = buildConfiguration(values);
+      formData.append("configuration", JSON.stringify(configuration));
+
+      if (values.use_custom_credentials) {
+        const credentials = buildCredentials(values);
+        formData.append("credentials", JSON.stringify(credentials));
+      } else {
+        formData.append("credentials", JSON.stringify({}));
+      }
+
+      formData.append("enabled", JSON.stringify(values.enabled ?? true));
+
+      // Send provider_id as an array for consistency with the action
+      formData.append("providers", JSON.stringify([values.provider_id]));
+    }
+
+    return formData;
+  };
+
+  const onSubmit = async (values: any) => {
+    const formData = buildFormData(values);
+
+    try {
+      let result;
+
+      if (isEditing && integration) {
+        result = await updateIntegration(integration.id, formData);
+      } else {
+        result = await createIntegration(formData);
+      }
+
+      if ("success" in result) {
+        toast({
+          title: "Success!",
+          description: `Security Hub integration ${isEditing ? "updated" : "created"} successfully.`,
+        });
+
+        if ("testConnection" in result) {
+          if (result.testConnection.success) {
+            toast({
+              title: "Connection test started!",
+              description:
+                "Connection test started. It may take some time to complete.",
+            });
+          } else if (result.testConnection.error) {
+            toast({
+              variant: "destructive",
+              title: "Connection test failed",
+              description: result.testConnection.error,
+            });
+          }
+        }
+
+        onSuccess();
+      } else if ("error" in result) {
+        const errorMessage = result.error;
+
+        toast({
+          variant: "destructive",
+          title: "Security Hub Integration Error",
+          description: errorMessage,
+        });
+      }
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : "An unexpected error occurred";
+
+      toast({
+        variant: "destructive",
+        title: "Connection Error",
+        description: `${errorMessage}. Please check your network connection and try again.`,
+      });
+    }
+  };
+
+  const renderStepContent = () => {
+    if (isEditingCredentials) {
+      // When editing credentials, show the credential type selector first
+      return (
+        <div className="space-y-4">
+          <RadioGroup
+            size="sm"
+            value={useCustomCredentials ? "custom" : "provider"}
```

Comment:
Maybe we can add an aria-label for this?

```suggestion
            aria-label="Credential Type"
            value={useCustomCredentials ? "custom" : "provider"}
```

Reply:
well seen thanks!